pandapipes/topology/create_graph.py

```diff
@@ -75,8 +64,38 @@ def create_nxgraph(net, include_pipes=True, include_valves=True, include_pumps=T
                     del mg._adj[b][i]  # networkx versions 2.0
 
     # remove out of service junctions
-    if not include_out_of_service:
+    if "junction" not in neglect_in_service_comps:
         for b in net.junction.index[~net.junction.in_service.values]:
             mg.remove_node(b)
 
-    return mg
\ No newline at end of file
+    return mg
+
+
+def add_branch_component(comp, mg, net, include_comp, neglect_in_service_comps, weight_col):
+    table_name = comp.table_name()
+    tab = get_edge_table(net, table_name, include_comp)
+
+    if tab is not None:
+        in_service_name = comp.active_identifier()
+        from_col, to_col = comp.from_to_node_cols()
+        indices, parameter, in_service = init_par(tab, table_name in neglect_in_service_comps,
+                                                  in_service_name)
+        indices[:, F_JUNCTION] = tab[from_col].values
+        indices[:, T_JUNCTION] = tab[to_col].values
+        if weight_col is not None:
+            parameter[:, WEIGHT] = net[table_name][weight_col].values
+        add_edges(mg, indices, parameter, in_service, net, table_name)
+
+
+def init_par(tab, neglect_in_service=False, in_service_col="in_service"):
+    n = tab.shape[0]
+    indices = np.zeros((n, 3), dtype=np.int)
+    indices[:, INDEX] = tab.index
+    parameters = np.zeros((n, 1), dtype=np.float)
+
+    if neglect_in_service:
+        return indices, parameters, np.ones(n, dtype=np.bool)
+    elif in_service_col in tab:
+        return indices, parameters, tab[in_service_col].values.copy()
+    else:
+        return indices, parameters
```

Comment:
Isn't it possible to get a more generalized version both useable for pandapower and pandapipes

Reply:
Probably... however, that requires a few changes in pandapower as well... we can see, if we can make them under the hood.

Reply:
Okay, maybe add a todo and add a issue to pandapower and pandapipes. Till then, I would merge the PR.